examples/conanfile/package_id/settings_and_options.rst

```diff
@@ -0,0 +1,145 @@
+
+.. code-block:: bash
+
+    $ conan create . --settings compiler.version=192
+    ...
+    pkg/1.0.0: Package '4f267380690f99b3ef385199826c268f63147457' created
+
+    $ conan create . --settings compiler.version=193
+    ...
+    pkg/1.0.0: Package 'c13a22a41ecd72caf9e556f68b406569547e0861' created
+
+
+However, there are cases when a setting should not influence the resulting package ID.
+An example of this could be when a project generates an executable binary as a compilation result. In that case, the value of the compiler
+used is not relevant for the package and we can remove its influence in the package ID like so:
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        settings = "os", "arch", "compiler", "build_type"
+
+        def build(self):
+            # self.settings.compiler value will be used here to compile
+
+        def package_id(self):
+            # Remove compiler setting from package ID
+            del self.info.settings.compiler
+
+Why not removing the setting from the `settings` attribute? Becase the compiler value is still needed in the `build()` method to perform the copilation of the executable.
+
+
+How options influence the package ID
+-------------------------------------
+
+Options are used to specify characteristics that are particular to a single recipe, contrasting with settings that generally remain consistent across recipes within a project.
+They are usually a set of particular characteristic a library executble or conan package may have.
+
+For example, a `shared` option is a very common option used in recipes that can produce shared libraries. However, it could not be a setting as not all recipes produce shared libraries.
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        options = {"shared": [True, False]}
+        default_options = {"shared": True}
+
+As in the previous case with settings, the different values of an option will incluence the package ID and therefore, generate different packages depending on it.
+
+.. code-block:: bash
+
+    $ conan create . --options shared=True
+    ...
+    pkg/1.0.0: Package '1744785cb24e3bdca70e27041dc5abd20476f947' created
+
+    $ conan create . --options shared=False
+    ...
+    pkg/1.0.0: Package '55c609fe8808aa5308134cb5989d23d3caffccf2' created
+
+In the same way, there might be options that are needed as input in a recipe to generate a package but that they shouldn't be taken into account in the package ID.
+To do so, the recipe can remove the option in the `pacakge_id()` method:
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        options = {"diagnostics_level": [0, 1, 2]}
+        default_options = {"diagnostics_level": 0}
```

Comment:
This is no longer recommended usage anymore, better use ``conf`` for this.

Removing options should be way less common nowadays.
I'd recommend documenting this possibility just as such "If you are using options and removing them from ``package_id``, consider using ``conf`` instead.

Reply:
Sure. I have left the example but recommended the conf usage at the end. Thanks!